### PR TITLE
PaymentNotification sdk

### LINKED ISSUE #262: arkd should notify clients for receiving transactions

### PR DESCRIPTION
**Pull request by sekulicd**
This adds implementation for new SDK ArkClient method PaymentNotification.

**Usage**:
Method returns Payment chan on which clients can listen for new payments for their pubkey.

**Implementation**:
PaymentNotification listen for round events using GetEventStream RPC, once RoundFinalizedEvent is noticed, it invokes GetRoundByID RPC which response is enriched with new proto msg:
```
message Payment {
  string txid = 1;
  uint32 vout = 2;
  bool spent = 3;
  bool pending = 4;
  uint32 amount = 5;
  string pubkey = 6;
}
``` 
and searches for payments related to provided pubkey.

This closes #262 

@tiero @bordalix please review.

### CODE REVIEW

**Comment by tiero** [comment]
I think if you push to origin, then people can try SDK before mergin to master ie. go get <>@commit_hash

**Comment by sekulicd** [comment]
I tested code by adding payment logging into example code, however there is an issue with current design, GetEventStream closes stream once event RoundFailed/RoundFinalized occurs.
This causes subscribing chan to close, [here](https://github.com/ark-network/ark/blob/3c8ab2803b2808b98f3b6bd486c550713bb82bc6/pkg/client-sdk/example/covenant/alice_to_bob.go#L244), which means user can listen for only one round.
@altafan any reason why we close stream in [here](https://github.com/sekulicd/ark/blob/ce6a4567ab2c47aaedcd20a20d22fa76a1f1de67/server/internal/interface/grpc/handlers/arkservice.go#L289)?

@tiero @bordalix

**Comment by tiero** [comment]
I assume the best commit to test is [this](https://github.com/ark-network/ark/commit/3c8ab2803b2808b98f3b6bd486c550713bb82bc6)

File: pkg/client-sdk/client.go [hunk 1]
```diff
@@ -229,6 +230,46 @@ func (a *arkClient) ping(
...
+
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case notify := <-eventsCh:
+			if notify.Err != nil {
+				return notify.Err
+			}
+
```
Comment (tiero) [inline pkg/client-sdk/client.go:247]: I think we shoudl also fire an event when a new pending payment is added?

**Comment by tiero** [comment]
merge/rebase from master

**Pull request closed without merging**